File: Math and Logic/Palindrome/Palindrome.py
def reverse(number):
    
    reversedNumber = 0

    while number > 0:

        reversedNumber = (reversedNumber * 10) + (number % 10)
        number = number // 10

    return reversedNumber

def isPalindrome(number):
    return  number == reverse(number)

number = int(input("Enter a number: ").strip())
result = isPalindrome(number)
print("Entered number is", "a palindrome." if result == True else "not a palindrome.")

File: Math and Logic/Palindrome/Palindrome.cpp
#include <iostream>

using namespace std;

int reverse(int number)
{

    int reversedNumber = 0;

    while (number > 0)
    {
        reversedNumber = (reversedNumber * 10) + (number % 10);
        number = (int)(number / 10);
    }

    return reversedNumber;
}

bool isPalindrome(int number)
{

    return number == reverse(number);
}

int main()
{

    int number;

    cout << "Enter a number: ";
    cin >> number;

    string result = (isPalindrome(number) == true) ? "a palindrome." : "not a palindrome.";
    cout << "Entered number is " << result;
    return 0;
}